doc/index.rst

```diff
@@ -0,0 +1,366 @@
+top of the browser window can be used to turn mask layers on and off.
+
+The :meth:`display1.setMaskPlaneColor` and
+:meth:`display1.setMaskTransparency` methods can be used to programmatically
+change the mask display. :meth:`display1.setMaskPlaneColor` must be used before
+the image is displayed, while the transparency may be changed at any time.
+
+.. code-block:: py
+    :name: mask-manipulation
+
+    display1.setMaskPlaneColor('DETECTED', afw_display.GREEN)
+    display1.mtv(calexp)
+    display1.setMaskTransparency(30)
+
+Image scale, zoom, pan
+----------------------
+
+The ``display1`` object includes methods for setting the image scale or
+stretch, the zoom and the pan position.
+
+.. code-block:: py
+    :name: scale-zoom-pan
+
+    display1.scale('log', -1, 10, 'sigma')
+    display1.zoom(4)
+    display1.pan(1064, 890)
+
+Overlaying symbols
+------------------
+
+The :meth:`display1.dot` method will overlay a symbol at a point.
+
+.. code-block:: py
+    :name: ff-dot
+
+    display1.dot('x', 1064, 890, size=8, ctype=afw_display.RED)
+
+interact() method not implemented
+---------------------------------
+
+The :meth:`display1.interact()` method is not implemented, due mainly to
+limitations of a browser window recognizing a keypress event.
+
+
+.. _lsst-display-firefly-installing:
+
+Installing lsst.display.firefly
+===============================
+
+Since `display_firefly` is not yet included in the `lsst_distrib` set of stack
+packages, this section outlines several installation scenarios.
+
+.. _lsst-display_firefly-eups-distrib-install:
+
+Installing with eups distrib install
+------------------------------------
+
+To check for published distributions of `display_firefly`, use
+
+.. code-block:: shell
+    :name: eups-distrib-list
+
+    eups distrib list display_firefly -s https://sw.lsstcorp.org/eupspkg
+
+This command will return the published versions of `display_firefly`, with
+the third item displayed as the EUPS version. Provide that version to
+`eups distrib install display_firefly`, e.g.:
+
+.. code-block:: shell
+    :name: eups-distrib-install
+
+    eups distrib install display_firefly master-g05bec400a4
+
+The version may not be compatible with the version of the stack you are
+using. This will be fixed when `display_firefly` is included in the
+`lsst_distrib` distribution.
+
+
+Developer installation from source code
+---------------------------------------
+
+Install using lsstsw
+^^^^^^^^^^^^^^^^^^^^
+
+If using `lsstsw` to develop with the stack, rebuild the package:
+
+.. code-block:: shell
+    :name: rebuild-firefly
+
+    rebuild display_firefly
+
+Note the build number NNNN that is output by the rebuild process, then:
+
+.. code-block:: shell
+    :name: tag-clone
+
+    eups tags --clone bNNNN current
+
+Install using Github and eups
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Set up the stack with `lsst_distrib` or an obs package such as `obs_sdss` -- just
+use something that already contains `lsst.afw`.
```

Comment:
This instruction shouldn't be needed.  `display_firefly` has a dependence on `afw` so will set it up if it can be found in the stack.

Reply:
I remember now, this part of the instructions use `setup -j` which won't bring in `afw`. I did find I needed to use this method inside a Docker container of the stack, where lsstsw was not available.

Reply:
You are saying you must use `setup -j`?  We should probably fix whatever problem is cause you to have to do that.

Reply:
Apologies, I just saw this comment. I've been using `setup -j` to allow the alternative of installing `firefly_client` and `ws4py` using pip. Those packages are in the stack as third-party packages, so the only ways I know to build them from the lsst repos are with rebuild in the lsstsw environment, or with eupspkg which looks like special magic.

I tried out the instructions inside a weekly stack Docker container. There's no lsstsw inside. There is eupspkg but it wants a newer git (1.8.4 or later). I installed git 2.15.1 as root and then the eupspkg instructions worked for `ws4py` and `firefly_client`. The order matters, so if I eupspkg-installed `ws4py`, and then `firefly_client`, then `setup` should work for `display_firefly` without the `-j` option.

I haven't had success with 'eups distrib install'. Inside the container, it want to recompile afw and that keeps failing for me. Maybe the right thing to do is to build-and-publish `display_firefly` with the weekly tag, so one could `eups distrib install` inside the Docker container without triggering a rebuild of `afw`? The rebuild of `afw` is what keeps failing for me.